subprojects/docs/src/docs/userguide/java.adoc

```diff
@@ -0,0 +1,453 @@
+
+The above is rarely sufficient, though. You will often need to adjust the compilation classpath, for example to include the production classes. And you will typically need to set up some dependencies. Here's a slightly more useful example that does both:
+
+++++
+<sample id="practicalIntegTest" dir="userguide/java/basic" title="Setting up working integration tests">
+    <sourcefile file="build.gradle" snippet="practical-integ-test-source-set"/>
+</sample>
+++++
+
+There are several important points to understand in the above example:
+
+ * `sourceSets.main.output` is a <<sec:file_collections,file collection>> of all the directories containing compiled production classes and resources
+ * `+=` allows you to append paths and collections of paths to `compileClasspath` and `runtimeClasspath`
+ * if you want to use the convention-based configurations, you _must_ declare the dependencies _after_ the new source set
+
+One thing to bear in mind is that source sets are geared around compilation and the processing of other resources. What this means for the integration test example is that we would still need to create a task to _run_ the tests. Here is an example for completeness sake:
+
+++++
+<sample id="integTestTask" dir="userguide/java/basic" title="Defining a working integration test task">
+    <sourcefile file="build.gradle" snippet="integ-test-task"/>
+</sample>
+++++
+
+Again, we're accessing a source set — `intTest` this time — to get the relevant information, i.e. where the compiled test cases are and what the classpath is for running those tests.
+
+The <<sec:some_source_set_examples,Java Plugin reference>> has some more examples of using source sets for things like creating JARs and generating Javadocs.
+
+
+[[sec:java_resources]]
+=== Managing resources
+
+Many Java projects make use of resources beyond source files, such as images, configuration files and localization data. Sometimes these files simply need to be packaged unchanged and sometimes they need to be processed as template files or in some other way. Either way, the Java Plugin adds a specific api:org.gradle.api.tasks.Copy[] task for each source set that handles the processing of its associated resources.
+
+The task's name follows the convention of `process__SourceSet__Resources` — or `processResources` for the `main` source set — and it will automatically copy any files in _src/[sourceSet]/resources_ to a directory that will be included in the production JAR. This target directory will also be included in the runtime classpath of the tests.
+
+Since `processResources` is an instance of the `Copy` task, you can perform any of the processing described in the _<<sec:copying_files,Working With Files>>_ chapter.
+
+[[sec:properties_files]]
+==== Java properties files
```

Comment:
Seems like the title or section content should mention that this is useful for creating reproducible builds.